python_logging_rabbitmq/handlers_oneway.py

```diff
@@ -151,22 +156,26 @@ def message_worker(self):
         while not self.stopping.is_set():
             try:
                 record, routing_key = self.queue.get(block=True, timeout=10)
-
-                if not self.connection or self.connection.is_closed or not self.channel or self.channel.is_closed:
-                    self.open_connection()
-
-                res = self.channel.basic_publish(
-                    exchange=self.exchange,
-                    routing_key=routing_key,
-                    body=record,
-                    properties=pika.BasicProperties(
-                        delivery_mode=2,
-                        headers=self.message_headers,
-                        content_type=self.content_type
+                
+                try:
+                    if not self.connection or self.connection.is_closed or not self.channel or self.channel.is_closed:
+                        self.open_connection()
+
+                    res = self.channel.basic_publish(
+                        exchange=self.exchange,
+                        routing_key=routing_key,
+                        body=record,
+                        properties=pika.BasicProperties(
+                            delivery_mode=2,
+                            headers=self.message_headers,
+                            content_type=self.content_type
+                        )
                     )
-                )
+                finally:
+                    if not hasattr(self, 'queue'):
```

Comment:
We want to call `task_done()` if there is a queue instance. Remove the `not` operator to be:

```python
if hasattr(self, 'queue'):
    self.queue.task_done()
```